service/matching/task_reader.go

```diff
@@ -149,6 +149,7 @@ dispatchLoop:
 				}
 				common.InterruptibleSleep(ctx, taskReaderOfferThrottleWait)
 			}
+			return ctx.Err()
```

Comment:
this is unnecessary since if ctx.Err() != nil, it'll break the outer loop also and then return. either way